--- a/mkspecs/android-clang/qmake.conf
+++ b/mkspecs/android-clang/qmake.conf
@@ -66,8 +66,9 @@
 QMAKE_CFLAGS_HIDESYMS   = -fvisibility=hidden
 QMAKE_CFLAGS_NEON       = -mfpu=neon
 
-QMAKE_LFLAGS_APP        = -Wl,--build-id=sha1 -Wl,--no-undefined -Wl,-z,noexecstack -shared
-QMAKE_LFLAGS_SHLIB      = -Wl,--build-id=sha1 -Wl,--no-undefined -Wl,-z,noexecstack -shared
+QMAKE_LFLAGS_APP        = -Wl,--build-id=sha1,--no-rosegment,--gc-sections,--exclude-libs=libgcc.a,--exclude-libs=libgcc_real.a -shared
+QMAKE_LFLAGS_SHLIB      = -Wl,--build-id=sha1,--no-rosegment,--gc-sections,--exclude-libs=libgcc.a,--exclude-libs=libgcc_real.a -shared
+DEFINES                += _FORTIFY_SOURCE=2
 QMAKE_LFLAGS_PLUGIN     = $$QMAKE_LFLAGS_SHLIB
 QMAKE_LFLAGS_NOUNDEF    = -Wl,--no-undefined
 QMAKE_LFLAGS_RPATH      = -Wl,-rpath=
--- a/mkspecs/features/android/default_pre.prf
+++ b/mkspecs/features/android/default_pre.prf
@@ -45,9 +45,11 @@
 
 equals(ANDROID_TARGET_ARCH, armeabi-v7a) | equals(ANDROID_TARGET_ARCH, armeabi) {
     CONFIG               += optimize_size
+    QMAKE_CFLAGS         += -mfpu=neon
     QMAKE_CFLAGS_DEBUG    = -g -marm -O0
     QMAKE_CFLAGS_RELEASE += -mthumb
     QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO += -mthumb
+    QMAKE_LINK           += -Wl,--exclude-libs,libunwind.a
 }
 
 QMAKE_CXXFLAGS_RELEASE_WITH_DEBUGINFO = $$QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO
