--- a/src/network/ssl/qsslsocket_openssl11.cpp	2018-09-18 21:56:40.444638484 +0300
+++ b/src/network/ssl/qsslsocket_openssl11.cpp	2018-09-18 21:57:54.844788847 +0300
@@ -127,16 +127,4 @@
     s_loadRootCertsOnDemand = true;
-#elif defined(Q_OS_UNIX) && !defined(Q_OS_DARWIN)
-    // check whether we can enable on-demand root-cert loading (i.e. check whether the sym links are there)
-    QList<QByteArray> dirs = unixRootCertDirectories();
-    QStringList symLinkFilter;
-    symLinkFilter << QLatin1String("[0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f][0-9a-f].[0-9]");
-    for (int a = 0; a < dirs.count(); ++a) {
-        QDirIterator iterator(QLatin1String(dirs.at(a)), symLinkFilter, QDir::Files);
-        if (iterator.hasNext()) {
-            s_loadRootCertsOnDemand = true;
-            break;
-        }
-    }
 #endif
 #endif // QT_CONFIG(library)
     // if on-demand loading was not enabled, load the certs now
