--- a/mkspecs/android-clang/qmake.conf	2019-06-12 23:59:14.000000000 +0300
+++ b/mkspecs/android-clang/qmake.conf	2019-08-08 01:30:07.504970680 +0300
@@ -24,10 +24,6 @@
     QMAKE_CFLAGS += -target i686-none-linux-android -mstackrealign
 else: equals(ANDROID_TARGET_ARCH, x86_64): \
     QMAKE_CFLAGS += -target x86_64-none-linux-android
-else: equals(ANDROID_TARGET_ARCH, mips): \
-    QMAKE_CFLAGS += -target mipsel-none-linux-android
-else: equals(ANDROID_TARGET_ARCH, mips64): \
-    QMAKE_CFLAGS += -target mips64el-none-linux-android
 
 QMAKE_CFLAGS += -gcc-toolchain $$NDK_TOOLCHAIN_PATH -fno-limit-debug-info
 
