--- a/mkspecs/features/qt_configure.prf	2019-06-13 07:18:18.000000000 +0300
+++ b/mkspecs/features/qt_configure.prf	2019-08-10 01:17:01.897550044 +0300
@@ -1241,7 +1241,8 @@
 
     test_base_out_dir = $$OUT_PWD/$$basename(QMAKE_CONFIG_TESTS_DIR)
     isEmpty(test) {
-        test_dir = $$test_base_out_dir/$$section(1, ".", -1)
+        test = $$section(1, ".", -1)
+        test_dir = $$test_base_out_dir/$$test
         test_out_dir = $$test_dir
         qtConfPrepareCompileTestSource($$1, $$test_dir)
     } else {
@@ -1283,6 +1284,10 @@
     # for platforms with multiple architectures (macOS, iOS, tvOS, watchOS),
     # make sure tests are only built for a single architecture
     qmake_configs += "single_arch"
+    
+    equals(test, openssl) {
+        qmake_configs += "ltcg"
+    }
 
     qmake_args += "\"CONFIG += $$qmake_configs\""
 
