--- a/configure	2018-06-15 10:29:31.000000000 +0300
+++ b/configure	2018-09-18 21:35:39.081990864 +0300
@@ -799,7 +799,7 @@
 
     if [ "$OPT_VERBOSE" = yes ]; then
         # Show the output of make
-        (cd "$outpath/qmake"; "$MAKE") || exit 2
+        (cd "$outpath/qmake"; "$MAKE" $MAKEOPTS) || exit 2
     else
         # Hide the output of make
         # Use bash to print dots, if we have it, and stdout is a tty.
@@ -807,9 +807,9 @@
             bash -c 'set -o pipefail
                 cd "$0/qmake"; "$1" | while read line; do
                     builtin echo -n .
-                done' "$outpath" "$MAKE" || exit 2
+                done' "$outpath" "$MAKE" $MAKEOPTS || exit 2
         else
-            (cd "$outpath/qmake"; "$MAKE" -s) || exit 2
+            (cd "$outpath/qmake"; "$MAKE" $MAKEOPTS -s) || exit 2
         fi
         echo "Done."
     fi
