diff --git a/mkspecs/common/clang.conf b/mkspecs/common/clang.conf
index dacd1539cf9..e03dc7de149 100644
--- a/mkspecs/common/clang.conf
+++ b/mkspecs/common/clang.conf
@@ -19,7 +19,8 @@ QMAKE_PCH_OUTPUT_EXT    = .pch
 QMAKE_CFLAGS_ISYSTEM             = -isystem
 QMAKE_CFLAGS_PRECOMPILE          = -x c-header -c ${QMAKE_PCH_INPUT} -o ${QMAKE_PCH_OUTPUT}
 QMAKE_CFLAGS_USE_PRECOMPILE      = -Xclang -include-pch -Xclang ${QMAKE_PCH_OUTPUT}
-QMAKE_CFLAGS_LTCG                = -flto
+QMAKE_CFLAGS_LTCG                = -flto=thin
+QMAKE_CFLAGS_LTCG_FATOBJECTS     = -flto
 QMAKE_CFLAGS_DISABLE_LTCG        = -fno-lto
 QMAKE_CFLAGS_GNUC99              = -std=gnu99
 QMAKE_CFLAGS_GNUC11              = -std=gnu11
@@ -27,6 +28,7 @@ QMAKE_CFLAGS_GNUC11              = -std=gnu11
 QMAKE_CXXFLAGS_PRECOMPILE        = -x c++-header -c ${QMAKE_PCH_INPUT} -o ${QMAKE_PCH_OUTPUT}
 QMAKE_CXXFLAGS_USE_PRECOMPILE    = $$QMAKE_CFLAGS_USE_PRECOMPILE
 QMAKE_CXXFLAGS_LTCG              = $$QMAKE_CFLAGS_LTCG
+QMAKE_CXXFLAGS_LTCG_FATOBJECTS   = $$QMAKE_CFLAGS_LTCG_FATOBJECTS
 QMAKE_CXXFLAGS_DISABLE_LTCG      = $$QMAKE_CFLAGS_DISABLE_LTCG
 QMAKE_CXXFLAGS_CXX11             = -std=c++11
 QMAKE_CXXFLAGS_CXX14             = -std=c++1y
