--- a/mkspecs/features/qt_configure.prf
+++ b/mkspecs/features/qt_configure.prf
@@ -1179,7 +1179,8 @@ defineTest(qtConfTest_compile) {
 
     test_base_out_dir = $$OUT_PWD/$$basename(QMAKE_CONFIG_TESTS_DIR)
     isEmpty(test) {
-        test_dir = $$test_base_out_dir/$$section(1, ".", -1)
+        test = $$section(1, ".", -1)
+        test_dir = $$test_base_out_dir/$$test
         test_out_dir = $$test_dir
         qtConfPrepareCompileTestSource($$1, $$test_dir)
     } else {
@@ -1222,6 +1223,12 @@ defineTest(qtConfTest_compile) {
     # make sure tests are only built for a single architecture
     qmake_configs += "single_arch"
 
+    ltcg {
+        equals(test, openssl) {
+            qmake_configs += "ltcg"
+        }
+    }
+
     qmake_args += "\"CONFIG += $$qmake_configs\""
 
     !$$host|!cross_compile {
